From 93c406885fedd6e98950dd4518b1dea6e7382643 Mon Sep 17 00:00:00 2001
From: Chandan Gera <cgera@codeaurora.org>
Date: Wed, 12 Oct 2016 21:26:02 +0530
Subject: [PATCH] QCamera2: Make Single camera as Standalone

Issue:
Even if single camera is detected HAL is
making it as Main camera whereas it should be
treated as STANALONE.

Fix:
Donot sort cameras if only one camera is detected.

Change-Id: I1d31a0762a63e76f6160e98e276c8a691d5225b0
CRs-Fixed: 1077031
---
 QCamera2/stack/mm-camera-interface/src/mm_camera_interface.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/QCamera2/stack/mm-camera-interface/src/mm_camera_interface.c b/QCamera2/stack/mm-camera-interface/src/mm_camera_interface.c
index 8537ac1..bf565a8 100644
--- a/QCamera2/stack/mm-camera-interface/src/mm_camera_interface.c
+++ b/QCamera2/stack/mm-camera-interface/src/mm_camera_interface.c
@@ -2734,7 +2734,9 @@ uint8_t get_num_of_cameras()
     g_cam_ctrl.num_cam = num_cameras;
 
     get_sensor_info();
-    sort_camera_info(g_cam_ctrl.num_cam);
+    if (g_cam_ctrl.num_cam > 1) {
+        sort_camera_info(g_cam_ctrl.num_cam);
+    }
     /* unlock the mutex */
     pthread_mutex_unlock(&g_intf_lock);
     LOGI("num_cameras=%d\n", (int)g_cam_ctrl.num_cam);
-- 
1.9.1

